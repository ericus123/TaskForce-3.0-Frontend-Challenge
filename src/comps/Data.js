/* eslint-disable react/prop-types */
import React from "react";
import {Row, Container, Col} from "react-bootstrap";
import { formatNumber } from "../helpers";
import "./styles.scss";
const Data = ({countryData,vaccineCountryData}) => {

    return (
        <Container className="data">

            <Row className=" data-row-1">
                <h1>{formatNumber(countryData.cases)}</h1>
                <span className="h5">Cummulatevely</span>
            </Row>
    
            <Row className="data-row-2 justify-content-center">
                <Col className="data-col m-auto">
                    <h1>-</h1>
                    <span>Tests</span>
                    <h3>{formatNumber(countryData.tests)}</h3>
                </Col>
                <Col className="data-col m-auto">
                    <h1>{formatNumber(countryData.todayCases)}</h1>
                    <span>Positive Cases</span>
                    <h3>{formatNumber(countryData.cases)}</h3>
                </Col>
                <Col className="data-col m-auto">
                    <h1>{formatNumber(countryData.todayCases)}</h1>
                    <span>Hospitalized</span>
                    <h3>{formatNumber(countryData.active)}</h3>
                </Col>
                <Col className="data-col m-auto">
                    <h1>{formatNumber(countryData.todayRecovered)}</h1>
                    <span>Recovered</span>
                    <h3>{formatNumber(countryData.recovered)}</h3>
                </Col>
             
                <Col className="data-col m-auto">
                    <h1>{formatNumber(countryData.todayDeaths)}</h1>
                    <span>Deaths</span>
                    <h3>{formatNumber(countryData.deaths)}</h3>
                </Col>
                {vaccineCountryData?.timeline ? 
                    <Col className="data-col m-auto">
                        <h1>{vaccineCountryData?.timeline[0].daily || "-"}</h1>
                        <span>Vaccinated</span>
                        <h3>{vaccineCountryData?.timeline[0].total || "-"}</h3>
                    </Col> : 
                    <Col className="data-col m-auto">  <h1>-</h1>
                        <span>Vaccinated</span>
                        <h3>-</h3></Col>
                  
                }
               
            </Row>

        </Container>
    );
};
export default Data;