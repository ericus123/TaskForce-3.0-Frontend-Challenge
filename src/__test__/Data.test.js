/* eslint-disable no-undef */
import React from "react";
import { shallow,configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Data from "../comps/Data";
import {Provider} from "react-redux"; 
import store from "../redux/store";
import toJson from "enzyme-to-json";
configure({adapter: new Adapter()});

it("renders country data without crashing", () => {
    const  countryData =
{
    "updated": 1629566946482,
    "country": "Rwanda",
    "countryInfo": {
        "_id": 646,
        "iso2": "RW",
        "iso3": "RWA",
        "lat": -2,
        "long": 30,
        "flag": "https://disease.sh/assets/img/flags/rw.png"
    },
    "cases": 82215,
    "todayCases": 0,
    "deaths": 1005,
    "todayDeaths": 0,
    "recovered": 45204,
    "todayRecovered": 0,
    "active": 36006,
    "critical": 43,
    "casesPerOneMillion": 6175,
    "deathsPerOneMillion": 75,
    "tests": 2285145,
    "testsPerOneMillion": 171622,
    "population": 13315025,
    "continent": "Africa",
    "oneCasePerPeople": 162,
    "oneDeathPerPeople": 13249,
    "oneTestPerPeople": 6,
    "activePerOneMillion": 2704.16,
    "recoveredPerOneMillion": 3394.96,
    "criticalPerOneMillion": 3.23
};
    const  vaccineCountryData = {
        "country": "Rwanda",
        "timeline": [
            {
                "total": 1344627,
                "daily": 0,
                "totalPerHundred": 0,
                "dailyPerMillion": 0,
                "date": "8/21/21"
            }
        ]
    };

    let wrapper = shallow(
        <Provider store={store}>
            <Data countryData={countryData} vaccineCountryData={vaccineCountryData}/>
        </Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot(); 
});

