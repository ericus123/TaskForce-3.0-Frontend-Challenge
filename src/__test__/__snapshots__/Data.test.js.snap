// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders country data without crashing 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Data
    countryData={
      Object {
        "active": 36006,
        "activePerOneMillion": 2704.16,
        "cases": 82215,
        "casesPerOneMillion": 6175,
        "continent": "Africa",
        "country": "Rwanda",
        "countryInfo": Object {
          "_id": 646,
          "flag": "https://disease.sh/assets/img/flags/rw.png",
          "iso2": "RW",
          "iso3": "RWA",
          "lat": -2,
          "long": 30,
        },
        "critical": 43,
        "criticalPerOneMillion": 3.23,
        "deaths": 1005,
        "deathsPerOneMillion": 75,
        "oneCasePerPeople": 162,
        "oneDeathPerPeople": 13249,
        "oneTestPerPeople": 6,
        "population": 13315025,
        "recovered": 45204,
        "recoveredPerOneMillion": 3394.96,
        "tests": 2285145,
        "testsPerOneMillion": 171622,
        "todayCases": 0,
        "todayDeaths": 0,
        "todayRecovered": 0,
        "updated": 1629566946482,
      }
    }
    vaccineCountryData={
      Object {
        "country": "Rwanda",
        "timeline": Array [
          Object {
            "daily": 0,
            "dailyPerMillion": 0,
            "date": "8/21/21",
            "total": 1344627,
            "totalPerHundred": 0,
          },
        ],
      }
    }
  />
</ContextProvider>
`;
